superset-frontend/src/explore/components/controls/DateFilterControl/components/CurrentCalendarFrame.tsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect } from 'react';
+import { t } from '@superset-ui/core';
+import { Radio } from 'src/components/Radio';
+import {
+  CURRENT_CALENDAR_RANGE_OPTIONS,
+  CURRENT_CALENDAR_RANGE_SET,
+} from 'src/explore/components/controls/DateFilterControl/utils';
+import {
+  CurrentCalendarRangeType,
+  CurrentCalendarWeek,
+  FrameComponentProps,
+} from '../types';
+
+export function CurrentCalendarFrame({ onChange, value }: FrameComponentProps) {
```

Comment:
**Issue**: The useEffect hook in CurrentCalendarFrame component may cause an infinite loop if the onChange function from props does not have a stable reference. <br> **Fix**: Ensure that the onChange function passed to CurrentCalendarFrame is wrapped in useCallback in the parent component to prevent potential infinite re-render loops. <br> **Code Suggestion**: 
 ```
 -export function CurrentCalendarFrame({ onChange, value }: FrameComponentProps) {
 +export function CurrentCalendarFrame({ onChange, value }: FrameComponentProps) {
 +  onChange = useCallback(onChange, []);
 ```